### PR TITLE
Fix for wrong output of successive AVG aggregations in GroupBy

### PR DESCRIPTION
**Pull request by tcm-marcel**
Fix for  #863. A wrong index was used to access the NullBitmap. 

The Bug only appeared when heaving an AVG aggregation, because AVGs get splitted up in three aggregations, a COUNT, a SUM and an internal AVG for the final result, that doesn't have a slot in the UpdateableStorage. Only in this case the storage index and the aggregation index differ and the bug appeared. 

Test cases have been added.

### CODE REVIEW

**Comment by tcm-marcel** [comment]
I just realized, that the bug also should have appeared with global aggregations, so the title and the commit message is a bit misleading. However, the fix solves both problems.

File: test/sql/aggregate_groupby_sql_test.cpp [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+
+    for (auto &result_row : ref_result) {
+      if (actual_result.count(result_row) == 0) {
+
+        EXPECT_EQ(ref_result, actual_result);
+        break;
+      }
+    }
+  }
+};
```
Comment (pmenon) [inline test/sql/aggregate_groupby_sql_test.cpp]: This is a useful function that other tests can use. Can you move it to a utility class?

File: test/sql/aggregate_groupby_sql_test.cpp [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+                                          "1|3|6|2",
+                                          "1|2|3|2",
+                                          "1|2|3|1",
+                                          "1|2|6|1"});
+
+
+  // free the database just created
+  txn = txn_manager.BeginTransaction();
+  catalog::Catalog::GetInstance()->DropDatabaseWithName(DEFAULT_DB_NAME, txn);
+  txn_manager.CommitTransaction(txn);
```
Comment (pmenon) [inline test/sql/aggregate_groupby_sql_test.cpp]: The tests are good. Does it make sense to move them into the aggregate_sql_test?

File: test/sql/aggregate_groupby_sql_test.cpp (hunk 1 above)
Comment (tcm-marcel) [reply test/sql/aggregate_groupby_sql_test.cpp]: I just talked with @pmenon and we decided to move the function as part of this PR. After this PR is merged I will adapt the test cases in #847 and #867 to use this utility function.

**Pull request merged**